src/V3Increments.cpp

```diff
@@ -0,0 +1,204 @@
+// -*- mode: C++; c-file-style: "cc-mode" -*-
+//*************************************************************************
+// DESCRIPTION: Verilator: Replace return/continue with jumps
+//
+// Code available from: https://verilator.org
+//
+//*************************************************************************
+//
+// Copyright 2003-2020 by Wilson Snyder. This program is free software; you
+// can redistribute it and/or modify it under the terms of either the GNU
+// Lesser General Public License Version 3 or the Perl Artistic License
+// Version 2.0.
+// SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0
+//
+//*************************************************************************
+// V3Increments's Transformations:
+//
+// XXX: TODO: Write some docs
+//
+//*************************************************************************
+
+#include "config_build.h"
+#include "verilatedos.h"
+
+#include "V3Global.h"
+#include "V3Increments.h"
+#include "V3Ast.h"
+
+#include <algorithm>
+#include <cstdarg>
+#include <vector>
+
+#include <iostream>
+
+//######################################################################
+
+class IncrementsVisitor : public AstNVisitor {
+private:
+    // STATE
+    int                 m_modIncrementsNum;  // Var name counter
+
+    // METHODS
+    void insertBefore(AstNode* placep, AstNode* newp) {
+        AstNRelinker linker;
+        placep->unlinkFrBack(&linker);
+        newp->addNext(placep);
+        linker.relink(newp);
+    }
```

Comment:
Instead of this can use standard function placep->addHereThisAsNext(newp).

Bigger problem though is you need to insert it above the previous statement.  V3Premit.cpp's insertBeforeStmt is an example of how to do this - complicated unfortunately.

Reply:
Thanks for the hint - I'm trying to copy `insertBeforeStmt` functionality to my class, and I see that it sets `m_funcp` pointer when it visits `AstCFunc` node, which is later checked in `startStatement` function to set `m_stmtp` pointer: 

`if (m_funcp) m_stmtp = nodep;`

but in my class the `AstCFunc` visit is never executed, which results in `m_funcp` being always NULL, and `m_stmtp` is not set, it looks like `AstCFunc` nodes are not present at the time I'm doing my processing, any idea how could I work around that?

Reply:
The problem is CFuncs are made in a later stage.  Probably version in V3Task is more suitable.